tests/linearGaussian_npse_test.py

```diff
@@ -1,300 +1,345 @@
+        x = linear_gaussian(theta, self.likelihood_shift, self.likelihood_cov)
+        return theta, x
+
+    def get_target_samples(self, num_samples: int, x_o):
+        if self.prior_type == "gaussian":
+            gt_posterior = true_posterior_linear_gaussian_mvn_prior(
+                x_o,
+                self.likelihood_shift,
+                self.likelihood_cov,
+                self.prior_mean,
+                self.prior_cov,
+            )
+            target_samples = gt_posterior.sample((num_samples,))
+        elif self.prior_type in {"uniform", None}:
+            target_samples = samples_true_posterior_linear_gaussian_uniform_prior(
+                x_o,
+                self.likelihood_shift,
+                self.likelihood_cov,
+                prior=self._get_default_prior(),
+                num_samples=num_samples,
+            )
+        else:
+            raise NotImplementedError(f"Unsupported prior type: {self.prior_type}")
+        return target_samples
+
+
+@dataclass(frozen=True)
+class NpseSamplingTestCase:
+    iid_method: str
+    sampling_method: str
+    num_trials: int
+
+    def __str__(self):
+        return f"{self.iid_method}-{self.sampling_method}-trials_{self.num_trials}"
+
+
+training_test_cases_gaussian = [
+    NpseTrainingTestCase(1, "gaussian", "vp"),
+    NpseTrainingTestCase(1, "gaussian", "ve"),
+    NpseTrainingTestCase(2, "gaussian", "ve"),
+    NpseTrainingTestCase(2, "gaussian", "vp"),
+    NpseTrainingTestCase(2, "gaussian", "subvp"),
+]
+
+training_test_cases_uniform = [
+    NpseTrainingTestCase(2, "uniform", "ve"),
+    NpseTrainingTestCase(2, "uniform", "vp"),
+    NpseTrainingTestCase(2, "uniform", "subvp"),
+    NpseTrainingTestCase(3, "uniform", "ve"),
+    NpseTrainingTestCase(3, "uniform", "vp"),
+    NpseTrainingTestCase(3, "uniform", "subvp"),
+]
+
+
+training_test_cases_all = training_test_cases_gaussian + training_test_cases_uniform
+
+sampling_test_cases_1_trial = [
+    NpseSamplingTestCase(iid, sampling, 1)
+    for iid, sampling in product(IID_METHODS, SAMPLING_METHODS)
+]
+
+sampling_test_cases_n_trials = [
+    NpseSamplingTestCase("fnpe", "sde", 3),
+    NpseSamplingTestCase("gauss", "sde", 3),
+    NpseSamplingTestCase("auto_gauss", "sde", 3),
+    NpseSamplingTestCase("jac_gauss", "sde", 3),
+    NpseSamplingTestCase("fnpe", "sde", 8),
+    NpseSamplingTestCase("gauss", "sde", 8),
+    NpseSamplingTestCase("auto_gauss", "sde", 8),
+    NpseSamplingTestCase("jac_gauss", "sde", 8),
+]
+
+sampling_test_cases_all = sampling_test_cases_1_trial + sampling_test_cases_n_trials
+
+
+def _get_regression_cases() -> List[Tuple[NpseTrainingTestCase, NpseSamplingTestCase]]:
+    """
+    # ToDO check if there is a bug for prior_type='uniform' and num_trial>1
+    # ToDO validate bug for combination sde_type='ode' and num_trial>1
+    # ToDO investigate non-determinism for prior_type=None, dim>= 2, ve, auto/jac_gauss
```

Comment:
I was wondering if we should mark these tests (non-deterministic and those that take too long) as `xfail` and set the timeout to, say, 20 seconds. This way, we would cover all possible combinations with the regression tests, and failing tests could be resolved later in the separate PRs.